spec/graphql/schema/subscription_spec.rb

```diff
@@ -469,20 +496,42 @@ def in_memory_subscription_count
       # Cause an authorized failure
       matz[:private] = true
       obj = OpenStruct.new(toot: { body: "Merry Christmas, here's a new Ruby version" }, user: matz)
-      SubscriptionFieldSchema.subscriptions.trigger(:toot_was_tooted, {handle: "matz"}, obj)
+      SubscriptionFieldSchema.subscriptions.trigger(:toot_was_tooted, {handle: "matz"}, obj, context: {})
       assert_equal 2, mailbox.size
       assert_equal ["Can't subscribe to private user ([\"tootWasTooted\"])"], mailbox.last["errors"].map { |e| e["message"] }
       # The subscription remains in place
       assert_equal 1, in_memory_subscription_count
     end
+
+    it "support dynamic schema with custom context" do
+      res = exec_query <<-GRAPHQL, context: { legacy_schema: true }
+      subscription {
+        tootWasTooted(handle: "matz") {
+          toot {
+            likesCount
+          }
+        }
+      }
+      GRAPHQL
+      assert_equal 1, in_memory_subscription_count
+      matz = SubscriptionFieldSchema::USERS["matz"]
+      obj = OpenStruct.new(toot: { likes_count: 42 }, user: matz)
+      SubscriptionFieldSchema.subscriptions.trigger(:toot_was_tooted, {handle: "matz"}, obj, context: {})
+
+      # Get 1 successful update
+      mailbox = res.context[:subscription_mailbox]
+      assert_equal 1, mailbox.size
+      update_payload = mailbox.first
+      assert_equal 42, update_payload["data"]["tootWasTooted"]["toot"]["likesCount"]
+    end
   end
 
   describe "applying `loads:`" do
     it "includes `as:` in the event topic" do
       assert_equal [], SubscriptionFieldSchema::InMemorySubscriptions::EVENT_REGISTRY.keys
       matz = SubscriptionFieldSchema::USERS["matz"]
       obj = OpenStruct.new(toot: { body: "I am a C programmer" }, user: matz)
-      SubscriptionFieldSchema.subscriptions.trigger(:toot_was_tooted, {handle: "matz"}, obj)
+      SubscriptionFieldSchema.subscriptions.trigger(:toot_was_tooted, {handle: "matz"}, obj, context: {})
```

Comment:
Out of curiousity, were these `context: {}`s neccessary? (I would have _expected_ this parameter's default value, `NullContext`, to be sufficient -- but maybe you added them for clarity?)

Reply:
Good question! `DuplicatedNameError` will be raised without all the `context: {}`s because `NullContext::NullWarden#visible_field?` always return `true` (Unlike `NullContext::NullWarden`). I did worry about whether adding all the `context: {}`s indicates a breaking change, but later realized the behavior (i.e. raising `DuplicatedNameError`) did not change before or after this PR.

Reply:
Thanks for clarifying that ... I think it'd be nice to _not_ have to pass an empty hash when we don't really need any context values, but I'm not sure the best way to accomplish that. I'll merge this and take a look at making that further change.